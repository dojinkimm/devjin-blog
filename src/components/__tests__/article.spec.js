import React from 'react'
import { create } from 'react-test-renderer'
import Article from '../Article'
import Headline from '../Article/Headline'
import Bodytext from '../Article/Bodytext'
const theme = require('../../theme/theme.json')

describe('Article Component', () => {
  it('renders correctly with title', () => {
    const tree = create(
      <Article theme={theme}>
        <Headline title="Test" theme={theme}></Headline>
      </Article>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly without title', () => {
    const tree = create(
      <Article theme={theme}>
        <Headline theme={theme}>
          <Bodytext theme={theme} body={mock_page} />
        </Headline>
      </Article>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

const mock_page =
  'function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }\n\nfunction _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }\n\nfunction _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }\n\n/* @jsx mdx */\nvar _frontmatter = {\n  "title": "Test"\n};\n\nvar makeShortcode = function makeShortcode(name) {\n  return function MDXDefaultShortcode(props) {\n    console.warn("Component " + name + " was not imported, exported, or provided by MDXProvider as global scope");\n    return mdx("div", props);\n  };\n};\n\nvar layoutProps = {\n  _frontmatter: _frontmatter\n};\nvar MDXLayout = "wrapper";\nreturn function MDXContent(_ref) {\n  var components = _ref.components,\n      props = _objectWithoutProperties(_ref, ["components"]);\n\n  return mdx(MDXLayout, _extends({}, layoutProps, props, {\n    components: components,\n    mdxType: "MDXLayout"\n  }), mdx("p", null, "test"));\n}\n;\nMDXContent.isMDXComponent = true;'
