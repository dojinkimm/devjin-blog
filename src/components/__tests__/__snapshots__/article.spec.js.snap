// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Component renders correctly with title 1`] = `
Array [
  <article
    className="article"
  >
    <h1>
      Test
    </h1>
    <style
      jsx={true}
    >
      
        h1 {
          font-size: 1.7em;
          margin: 0 0 40px 0;
          animation-name: headlineEntry;
          animation-duration: 1s;

          :global(span) {
            font-weight: 400;
            display: block;
            font-size: 0.5em;
            letter-spacing: 0;
            margin: 0 0 5px 0;
          }

          :global(svg) {
            height: 0.75em;
            fill: #2c3e50;
          }
        }

        @keyframes headlineEntry {
          from {
            opacity: 0.5;
          }
          to {
            opacity: 1;
          }
        }

        @from-width tablet {
          h1 {
            font-size: calc(2em * 1.2);
          }
        }

        @from-width desktop {
          h1 {
            font-size: calc(2em * 1.4);
          }
        }
      
    </style>
  </article>,
  <style
    jsx={true}
  >
    
        .article {
          padding: 20px;
          margin: 0 auto;
        }
        @from-width tablet {
          .article {
            padding: calc(20px) calc(20px * 2);
            max-width: 650px;
          }
        }
        @from-width desktop {
          .article {
            padding: calc(20px * 2 + 90px) 0 calc(20px * 2);
            max-width: 750px;
          }
        }
      
  </style>,
]
`;

exports[`Article Component renders correctly without title 1`] = `
Array [
  <article
    className="article"
  >
    <h1>
      <div
        className="bodytext"
      >
        <p>
          test
        </p>
      </div>
      <style
        jsx={true}
      >
        
        .bodytext {
          animation-name: bodytextEntry;
          animation-duration: 1s;

          :global(h2),
          :global(h3) {
            margin: 1.5em 0 1em;
          }

          :global(h2) {
            line-height: 1.2;
            font-size: 1.35em;
          }

          :global(h3) {
            font-size: 1.1em;
            line-height: 1.3;
          }

          :global(p) {
            font-size: 1.1em;
            line-height: 1.6;
            margin: 0 0 1.5em;
          }
          :global(ul) {
            list-style: circle;
            margin: 0 0 1.5em;
            padding: 0 0 0 1.5em;
          }
          :global(li) {
            margin: 0.7em 0;
            line-height: 1.5;
          }
          :global(a) {
            font-weight: 600;
            color: #2c3e50;
            text-decoration: underline;
          }
          :global(a.gatsby-resp-image-link) {
            border: 0;
            display: block;
            margin: 2.5em 0;
            border-radius: 10px;
            overflow: hidden;
            border: 1px solid #ecebea;
          }
          :global(code.language-text) {
            background: #ecebea;
            text-shadow: none;
            color: inherit;
            padding: 0.1em 0.3em 0.2em;
            border-radius: 0.1em;
          }
        }

        @keyframes bodytextEntry {
          from {
            opacity: 0;
          }
          to {
            opacity: 1;
          }
        }
      
      </style>
    </h1>
    <style
      jsx={true}
    >
      
        h1 {
          font-size: 1.7em;
          margin: 0 0 40px 0;
          animation-name: headlineEntry;
          animation-duration: 1s;

          :global(span) {
            font-weight: 400;
            display: block;
            font-size: 0.5em;
            letter-spacing: 0;
            margin: 0 0 5px 0;
          }

          :global(svg) {
            height: 0.75em;
            fill: #2c3e50;
          }
        }

        @keyframes headlineEntry {
          from {
            opacity: 0.5;
          }
          to {
            opacity: 1;
          }
        }

        @from-width tablet {
          h1 {
            font-size: calc(2em * 1.2);
          }
        }

        @from-width desktop {
          h1 {
            font-size: calc(2em * 1.4);
          }
        }
      
    </style>
  </article>,
  <style
    jsx={true}
  >
    
        .article {
          padding: 20px;
          margin: 0 auto;
        }
        @from-width tablet {
          .article {
            padding: calc(20px) calc(20px * 2);
            max-width: 650px;
          }
        }
        @from-width desktop {
          .article {
            padding: calc(20px * 2 + 90px) 0 calc(20px * 2);
            max-width: 750px;
          }
        }
      
  </style>,
]
`;
